### PR TITLE
fix(koa): sanitize protocol-relative redirects

### PR DESCRIPTION
**Pull request by killagu**
## Summary
- sanitize ambiguous ctx.redirect() targets that browsers may treat as protocol-relative URLs
- keep existing absolute http(s) formatting behavior
- add regression coverage for //host, /\host, and mixed slash/backslash variants

## Test
- pnpm exec oxfmt --check packages/koa/src/response.ts packages/koa/test/response/redirect.test.ts
- pnpm exec oxlint --type-aware packages/koa/src/response.ts packages/koa/test/response/redirect.test.ts
- pnpm --filter @eggjs/koa run typecheck
- pnpm exec vitest run packages/koa/test/response/redirect.test.ts --bail 1 --retry 0 --testTimeout 20000 --hookTimeout 20000

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved redirect normalization to sanitize and consistently handle a wider range of URL patterns (protocol-relative //, backslash-prefixed, mixed slash/backslash, double-slash), preventing unsafe redirects and improving stability.

* **Tests**
  * Added tests covering normalization of malformed and edge-case redirect inputs and ensuring redirects set expected status without throwing.

[Review Change Stack](https://app.coderabbit.ai/change-stack/eggjs/egg/pull/5951?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/koa/src/response.ts [hunk 1]
```diff
@@ -18,6 +18,23 @@ import type { Application } from './application.ts';
...
+  }
+  if (url.startsWith('/\\')) {
+    return `/%5C${url.slice(2)}`;
+  }
+  if (/^[\\/]{2,}/.test(url)) {
+    return `/%2F${url.slice(2)}`;
+  }
+  if (url.startsWith('\\')) {
+    return `/%5C${url.slice(1)}`;
+  }
```
[the remark this answers is not in the document]
Comment (killagu) [reply packages/koa/src/response.ts:40]: Addressed in f5c17f1 by removing the regex branch and using explicit leading-character checks for the slash/backslash combinations.
